File: esp32_project/close_daily.php
<?php
// Configuración de la conexión a la base de datos
$servername = "localhost";  // Cambia si es necesario
$username = "root";   // Cambia por tu usuario de la base de datos
$password = "39090169"; // Cambia por tu contraseña de la base de datos
$dbname = "esp32_report"; // Cambia por el nombre de tu base de datos

// Crear la conexión
$conn = new mysqli($servername, $username, $password, $dbname);

// Verificar si la conexión tuvo éxito
if ($conn->connect_error) {
    die("Conexión fallida: " . $conn->connect_error);
}

// Obtener la fecha y hora actual
$fecha_actual = date('Y-m-d H:i:s');

// Insertar la fecha actual en la tabla daily_closes
$sql = "INSERT INTO daily_closes (close_date) VALUES ('$fecha_actual')";

if ($conn->query($sql) === TRUE) {
    echo "Cierre diario registrado correctamente: $fecha_actual";
} else {
    echo "Error al registrar el cierre: " . $conn->error;
}

// Cerrar la conexión
$conn->close();
?>

File: esp32_project/close_daily.py
import mysql.connector
import time
from datetime import datetime

def subir_fecha():
    # Conectar a la base de datos MySQL
    conexion = mysql.connector.connect(
        host='localhost',
        user='root',          # Reemplaza con tu usuario de MySQL
        password='39090169',  # Reemplaza con tu contraseña de MySQL
        database='esp32_report'  # Reemplaza con el nombre de tu base de datos
    )
    cursor = conexion.cursor()

    # Obtener la fecha y hora actual
    fecha_actual = datetime.now().strftime('%Y-%m-%d %H:%M:%S')

    # Insertar la fecha actual en la tabla daily_closes
    cursor.execute('INSERT INTO daily_closes (close_date) VALUES (%s)', (fecha_actual,))
    
    # Guardar (commit) los cambios y cerrar la conexión
    conexion.commit()
    cursor.close()
    conexion.close()

    print(f"Fecha {fecha_actual} subida a la tabla daily_closes.")

# Ejecutar el script cada 24 horas
while True:
    # Obtener la hora actual
    hora_actual = datetime.now().strftime('%H:%M')
    
    # Comprobar si es exactamente 23:59
    if hora_actual == '23:59':
        subir_fecha()
        time.sleep(60)  # Esperar 1 minuto para evitar múltiples inserciones en el mismo minuto
    
    time.sleep(10)  # Esperar 10 segundos antes de volver a comprobar la hora
